### PR TITLE
fix the controls dispose issue causing an exeption in the unload

### PR DESCRIPTION
**Pull request by giakas**

### CODE REVIEW

File: packages/web-components/src/player-component/player.class.ts [hunk 1]
```diff
@@ -332,12 +331,12 @@ export class PlayerWrapper {
 
         // Remove timeline
         this.removeTimelineComponent();
-        // await this.controls.destroy();
 
         if (this.isLoaded) {
             this.video.pause();
             this.video.src = '';
             await this.player?.unload();
+            await this.controls.destroy();
```
Comment (RaymondCheng) [inline packages/web-components/src/player-component/player.class.ts:339]: Maybe this.controls = undefined for extra safety?

File: packages/web-components/src/player-component/player.class.ts [hunk 2]
```diff
@@ -332,12 +331,12 @@ export class PlayerWrapper {
...
         }
         this.isLoaded = false;
```
Comment (RaymondCheng) [inline packages/web-components/src/player-component/player.class.ts:341]: You can tuck this into the if-statement block above since it is only needed if this.isLoaded was true, otherwise it is already false.

File: packages/web-components/src/player-component/player.class.ts [hunk 3]
```diff
@@ -683,16 +682,16 @@ export class PlayerWrapper {
                 const videoCurrentTime = video.currentTime; // Lock in the current value
                 const videoBufferedEnd = videoBuffered.length > 0 ? videoBuffered[videoBuffered.length - 1].end : videoCurrentTime - 1;
                 if (videoBufferedEnd - MAX_LATENCY_WINDOW > videoCurrentTime) {
-                // TODO: To account for the time that elapses during the trickplay portion, we need to add some additional trickplay duration
-                // in order to get latency as low as possible. The slower the trickplay speed, and the higher the latency, the higher this
-                // additional value will have to be. For now we use a flat value.
-                    const additionalDur= 2000; // in milliseconds
+                    // TODO: To account for the time that elapses during the trickplay portion, we need to add some additional trickplay duration
```
Comment (RaymondCheng) [inline packages/web-components/src/player-component/player.class.ts:685]: I don't have a problem with this whitespace-only change but I am wondering how it got through. I guess lint doesn't check for indent? Otherwise the offending PR would have failed validation. In an ideal world this would be a separate commit and the PR would be completed using rebase and fast forward but time crunch being what it is...

File: packages/web-components/src/player-component/player.class.ts (hunk 1 above)
Comment (giakas) [reply packages/web-components/src/player-component/player.class.ts:339]: done

File: packages/web-components/src/player-component/player.class.ts (hunk 3 above)
Comment (giakas) [reply packages/web-components/src/player-component/player.class.ts:685]: we use prettier in our team, so on save the file auto formats with correct formatting. I can checkin the rules and recommended vsCode extensions in a later PR, so that everyone is using the same tools and rules.

**Review by RaymondCheng — APPROVED** [review]
